invoices/invoiceregistry.go

```diff
@@ -692,8 +717,28 @@ func (i *InvoiceRegistry) processKeySend(ctx invoiceUpdateCtx) error {
 	_, err = i.AddInvoice(invoice, ctx.hash)
 	if err != nil && err != channeldb.ErrDuplicateInvoice {
 		return err
+	} else if err == channeldb.ErrDuplicateInvoice {
+		dbInvoice, err := i.LookupInvoice(ctx.hash)
```

Comment:
I think not inside the lock, but no problem? Or (very edgy) can it be a problem if invoice was just deleted before this call?

Reply:
Not sure if we need a lock here given that it'll be serialized during DB access.

Reply:
if the invoice gets deleted between `AddInvoice` and `LookupInvoice` then we'll just fail the HTLC due to the error case immediately below